### ISSUE TITLE
[🐛 Bug]: Error when using different node versions

### ISSUE THREAD

**Issue by TamsilAmani** [SeleniumHQ/selenium#10970]
### What happened?

_Issue 1 is based on_ https://github.com/nightwatchjs/nightwatch/issues/3337

**_ISSUE 1:_**
On node v10 (10.24.1), when I run the selenium code, I get the error `SyntaxError: Unexpected token =` in file `virtual_authenticator.js`. The same script runs without any error in v12 (12.22.12). 
I am not able to test it on v11. 

**_ISSUE 2:_** 
There is another issue which occurs in the same selenium code. On node v18 (18.7.0) the code is not able to start geckodriver and it gives the error `Error: Timed out waiting for the WebDriver server at http://localhost:51846/`. The same code works without any error (i.e. it starts the geckodriver) in v16 (16.16.0)

This looks like some version problem with nodejs. I'm troubleshooting it from my end. Any leads would be helpful.
CC: @harsha509 

### How can we reproduce the issue?

```shell
Step 1: Set your nodejs version with your version manager. I use nvm.
Step 2: Run the following selenium code.

const {Builder} = require('selenium-webdriver');

(async function helloSelenium() {
    let driver = await new Builder().forBrowser('firefox').build();

    await driver.get('https://selenium.dev');

    await driver.quit();
})();
```

### Relevant log output

```shell
ISSUE 1:

/Users/tamsilamani/Desktop/BrowserStack/seleniumhq.github.io/examples/javascript/node_modules/selenium-webdriver/lib/virtual_authenticator.js:26
  Protocol = {
           ^

SyntaxError: Unexpected token =
    at Module._compile (internal/modules/cjs/loader.js:723:23)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:789:10)
    at Module.load (internal/modules/cjs/loader.js:653:32)
    at tryModuleLoad (internal/modules/cjs/loader.js:593:12)
    at Function.Module._load (internal/modules/cjs/loader.js:585:3)
    at Module.require (internal/modules/cjs/loader.js:692:17)
    at require (internal/modules/cjs/helpers.js:25:18)
    at Object.<anonymous> (/Users/tamsilamani/Desktop/BrowserStack/seleniumhq.github.io/examples/javascript/node_modules/selenium-webdriver/lib/webdriver.js:40:20)



ISSUE 2:

/Users/tamsilamani/Desktop/BrowserStack/seleniumhq.github.io/examples/javascript/node_modules/selenium-webdriver/http/util.js:88
        reject(Error('Timed out waiting for the WebDriver server at ' + url))
               ^

Error: Timed out waiting for the WebDriver server at http://localhost:51846/
    at onError (/Users/tamsilamani/Desktop/BrowserStack/seleniumhq.github.io/examples/javascript/node_modules/selenium-webdriver/http/util.js:88:16)
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
```

### Operating System

macOS Monterey

### Selenium version

Javascript 4.4.0

### What are the browser(s) and version(s) where you see this issue?

Firefox 103.0.2

### What are the browser driver(s) and version(s) where you see this issue?

GeckoDriver 0.31.0

**Comment by harsha509** [comment]
HI @TamsilAmani ,

 InstanceOf  doesn't work as expected in node 18!

Ex: insatnceOf in ff `createSession`

I have a fix ready for Firefox issue as I rewrote firefox.js a bit

need to debug for virtual_authenticator.js with node 12 and node 14.

For JS we had version set for node 14 AFAIK!

**Comment by harsha509** [comment]
Hi @TamsilAmani,

About issue 1,

I don't think we should support node 10. It is deprecated long ago and `selenium-webdriver` should target LTS and stable version and <LTS should be mostly unsupported!

**Comment by TamsilAmani** [comment]
Thank you for the commit. On issue 1, I agree with you on that @harsha509.

**Comment by TamsilAmani** [comment]
ISSUE 1 is occuring because:

1) In commit https://github.com/SeleniumHQ/selenium/commit/1c2240df9d767631c15334434c34e86269936296, in file `virtual_authenticator.js`, at line https://github.com/SeleniumHQ/selenium/blob/1c2240df9d767631c15334434c34e86269936296/javascript/node/selenium-webdriver/lib/virtual_authenticator.js#L24-L36, declares Protocol and Transport as static and within the class. Maybe node v10 does not support this approach.

2) The solution which I found, and which is already merged, is in commit https://github.com/SeleniumHQ/selenium/commit/b8ede333beecc3b4a543f09cc0b3315c66f2d488, at https://github.com/SeleniumHQ/selenium/blob/b8ede333beecc3b4a543f09cc0b3315c66f2d488/javascript/node/selenium-webdriver/lib/virtual_authenticator.js#L24-L44, which makes Protocol and Transport const and also declares them outside the class.


Hence I believe we have to make a new release which includes commit mentioned on point (2) to solve for Issue 1 @AutomatedTester

**Comment by harsha509** [comment]
As discussed, I've set the min node version to the least Maintainance LTS with security updates. 
if the Current LTS is 16(N), Maintainance LTS with security updates will be N-2, and stable will be N+2

This change is done with commit https://github.com/SeleniumHQ/selenium/commit/024d863a56e57bd32a1b133044266ec9d51dbbf8 !

Closing this as all issues are resolved and fixes will be available in 4.5.0!
